src/hb-ot-name.cc

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2018  Ebrahim Byagowi
+ *
+ *  This is part of HarfBuzz, a text shaping library.
+ *
+ * Permission is hereby granted, without written agreement and without
+ * license or royalty fees, to use, copy, modify, and distribute this
+ * software and its documentation for any purpose, provided that the
+ * above copyright notice and the following two paragraphs appear in
+ * all copies of this software.
+ *
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE TO ANY PARTY FOR
+ * DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES
+ * ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN
+ * IF THE COPYRIGHT HOLDER HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ *
+ * THE COPYRIGHT HOLDER SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING,
+ * BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE.  THE SOFTWARE PROVIDED HEREUNDER IS
+ * ON AN "AS IS" BASIS, AND THE COPYRIGHT HOLDER HAS NO OBLIGATION TO
+ * PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+ */
+
+#include "hb.hh"
+
+#include "hb-ot-face.hh"
+#include "hb-ot-name-table.hh"
+
+static inline const OT::name& _get_name (hb_face_t *face)
+{
+  if (unlikely (!hb_ot_shaper_face_data_ensure (face))) return Null(OT::name);
+  return *hb_ot_face_data (face)->name;
+}
+
+/**
+ * hb_ot_name_entries_count:
+ * @face: #hb_face_t
+ * @name_index:
+ * @start_offset:
+ * @languages_length: (in/out)
+ * @languages: (out)
+ *
+ * Return value:
+ *
+ * Since: REPLACEME
+ **/
+unsigned int
+hb_ot_name_get_languages (hb_face_t    *face,
+			  unsigned int  name_index,
+			  unsigned int  start_offset,
+			  unsigned int *languages_length,
+			  unsigned int *languages)
+{
+  const OT::name& table = _get_name (face);
+
+  unsigned int matched = 0;
+  for (unsigned int i = 0; i < table.count; i++)
+  {
+    const OT::NameRecord& record = table.get_record (i);
+    if (record.nameID == name_index)
+    {
+      if (matched >= start_offset)
+	languages[matched - start_offset] = record.languageID;
+      matched++;
+    }
+  }
+  if (languages_length) *languages_length = matched - start_offset;
+  return matched;
+}
+
+/**
+ * hb_ot_name_entries_count:
+ * @face: #hb_face_t
+ * @name_id:
+ * @language_id:
+ * @buffer_length: (in/out)
+ * @buffer: (out)
+ *
+ * Return value: The string buffer size
+ *
+ * Since: REPLACEME
+ **/
+unsigned int
+hb_ot_name_get_string (hb_face_t    *face,
+		       unsigned int  name_id,
+		       unsigned int  language_id,
```

Comment:
We need mapping from `hb_language_t`.

Really, there's quite some infrastructure to build before this API can be implemented.